lints/lint_rsa_algid_null_params.go

```diff
@@ -0,0 +1,103 @@
+/*
+ * ZLint Copyright 2019 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package lints
+
+import (
+	"bytes"
+	"crypto/rsa"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"fmt"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+type rsaAlgIDNullParams struct{}
+
+// Initialize is a NOP.
+func (l *rsaAlgIDNullParams) Initialize() error {
+	return nil
+}
+
+// CheckApplies returns true for certificates with an RSA public key.
+func (l *rsaAlgIDNullParams) CheckApplies(c *x509.Certificate) bool {
+	_, ok := c.PublicKey.(*rsa.PublicKey)
+	return ok && c.PublicKeyAlgorithm == x509.RSA
```

Comment:
perhaps checking 

`c.PublicKeyAlgorithmOID.Equal(rsaEncryptionOID)`

is slightly more future-proof as RFC4055 has some additional rules for pub keys with RSA-PSS identifiers and someone might use x509.RSA as a blanket RSA categorisation in the future.

Reply:
@tadukurow That seems sensible but I think it would be best if you could open a separate issue for that feedback. I took this approach based on what was done elsewhere in the codebase:

```
lints/lint_rsa_exp_negative.go
32:	return ok && c.PublicKeyAlgorithm == x509.RSA

lints/lint_old_sub_cert_rsa_mod_less_than_1024_bits.go
33:	return ok && c.PublicKeyAlgorithm == x509.RSA && !util.IsCACert(c) && endDate.Before(util.NoRSA1024Date)

lints/lint_rsa_public_exponent_too_small.go
37:	return ok && c.PublicKeyAlgorithm == x509.RSA

lints/lint_rsa_mod_not_odd.go
38:	return ok && c.PublicKeyAlgorithm == x509.RSA

lints/lint_rsa_mod_less_than_2048_bits.go
36:	return ok && c.PublicKeyAlgorithm == x509.RSA && c.NotAfter.After(util.NoRSA1024Date.Add(-1))

lints/lint_old_root_ca_rsa_mod_less_than_2048_bits.go
37:	return ok && c.PublicKeyAlgorithm == x509.RSA && util.IsRootCA(c) && issueDate.Before(util.NoRSA1024RootDate)

lints/lint_rsa_public_exponent_not_odd.go
37:	return ok && c.PublicKeyAlgorithm == x509.RSA

lints/lint_rsa_public_exponent_not_in_range.go
42:	return ok && c.PublicKeyAlgorithm == x509.RSA

lints/lint_rsa_mod_factors_smaller_than_752_bits.go
37:	return ok && c.PublicKeyAlgorithm == x509.RSA
```

I think it would make sense to adjust some (all?) of those other lints at the same time.

Reply:
Discussion on this feedback ended up un-threading, consensus is that this particular lint deserves the more robust OID check. The other referenced lints are more generic.